File: frontend/src/types/prediction.ts
export interface Prediction {
  id: number;
  employee_id: string;
  risk_score: number;
  threat_level: "LOW" | "MEDIUM" | "HIGH" | "CRITICAL" | string;
  confidence?: number | null;
  explanation: Record<string, unknown>;
  model_version: string;
  created_at: string;
}

export interface PredictionRequest {
  employee_id: string;
  time_window_start?: string;
  time_window_end?: string;
}

export interface BatchPredictionRequest {
  employee_ids: string[];
}

File: backend/app/schemas/prediction.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict

__all__ = ["PredictionRequest", "PredictionResponse", "BatchPredictionRequest"]


class PredictionRequest(BaseModel):
    employee_id: str
    context: dict[str, Any] | None = None
    time_window_start: datetime | None = None
    time_window_end: datetime | None = None


class PredictionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    employee_id: str
    risk_score: float
    threat_level: str
    confidence: float | None = None
    explanation: dict[str, Any] = {}
    model_version: str
    created_at: datetime


class BatchPredictionRequest(BaseModel):
    employee_ids: list[str]
